tests/ceph_volume/test_redeploy_collocated_and_non_collocated_osd.py

```diff
@@ -0,0 +1,126 @@
+from json import loads
+
+from ceph.waiter import WaitUntil
+from cephci.utils.configs import get_configs
+from cli.cephadm.cephadm import CephAdm
+from cli.exceptions import OperationFailedError
+from cli.utilities.utils import (
+    create_yaml_config,
+    get_lvm_on_osd_container,
+    get_running_containers,
+)
+
+
+def _get_osd_db_id(osd_ids, lvm_list):
+    """Get OSD id and associated devices name
+    Args:
+        osd_ids (list): list of osd ids
+        lvm_list (list): list of lvm
+    return (dict): osd id associated with device
+    """
+    osds = {}
+    for id in osd_ids:
+        for item in lvm_list.get(id):
+            if item.get("type") == "db":
+                osds[id] = item["devices"]
+    return osds
+
+
+def run(ceph_cluster, **kw):
+    """Re-deploy collocated and non-collocated OSDs with wrong dedicated DB size"""
+
+    # Get configs
+    get_configs()
+    config = kw.get("config")
+
+    # Get the installer and OSD nodes
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    osd_nodes = ceph_cluster.get_nodes(role="osd")
+
+    # Create spec for unmanaged OSDs
+    specs = config.get("spec")
+    specs["unmanaged"] = "true"
+    file = create_yaml_config(node=installer, config=specs)
+
+    # Update OSD as unmanged
+    c = {"pos_args": [], "input": file}
```

Comment:
Same as below

Reply:
Yes, it's mandatory paramter.

Reply:
No, it's not mandatory refer - https://github.com/red-hat-storage/cephci/blob/master/cli/ceph/orch/orch.py#L53

Try without this key

Reply:
@vamahaja, yes, I have did try it. It does fail without pos_args. I'll try again for your assurange.

Reply:
@vamahaja It does complain, without pos_arags:
``` 2024-02-13 20:39:37,218 (cephci.cli.utilities.operations) [ERROR] - cephci.run.py:805 - 'pos_args'
Traceback (most recent call last):
  File "run.py", line 789, in run
    rc = test_mod.run(
  File "/home/aramteke/Work/Ceph/Automation/cephci/tests/ceph_volume/test_ceph_volume_redeploy.py", line 93, in run
    CephAdm(nodes=installer, mount=file).ceph.orch.apply(**c)
  File "/home/aramteke/Work/Ceph/Automation/cephci/cli/ceph/orch/orch.py", line 73, in apply
    for arg in kw.pop("pos_args"):
KeyError: 'pos_args'
2024-02-13 20:39:37,22 ```